### PR TITLE
fix(useHookAtTopLevel): recognize hook property in member expression calls

### PR DESCRIPTION
**Pull request by ksnyder9801**
## Summary
this is needed for parity with eslint-plugin-react-hooks. specifically when using hook APIs like [trpc](https://trpc.io/docs/client/react/useQuery) or simply React.useEffect instead of useEffect, the current lint rule will fail to catch rules-of-hooks violations.

## Test Plan
updated test case

### CODE REVIEW

**Review by ematipico — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -59,6 +59,8 @@ our [guidelines for writing a good changelog entry](https://github.com/biomejs/b
 
 - [noEmptyInterface](https://biomejs.dev/linter/rules/no-empty-interface/) now ignores empty interfaces in ambient modules ([#3110](https://github.com/biomejs/biome/issues/3110)). Contributed by @Conaclos
 
+- [useHookAtTopLevel](https://biomejs.dev/linter/rules/use-hook-at-top-level/) now recognizes properties named as hooks like `foo.useFoo()`.
```
Comment (arendjr) [inline CHANGELOG.md:62]: Feel free to add a “Contributed by” note ;)

**Review by arendjr — APPROVED** [review]
Thanks!

**Comment by ematipico** [comment]
Can you open a new issue?

**Comment by ipanasenko** [comment]
Yes, sure: https://github.com/biomejs/biome/issues/3337

**Pull request merged**